tflog/provider.go

```diff
@@ -104,5 +130,142 @@ func Error(ctx context.Context, msg string, additionalFields ...map[string]inter
 		// so just making this a no-op is fine
 		return
 	}
-	logger.Error(msg, hclogutils.MapsToArgs(additionalFields...)...)
+
+	additionalArgs, shouldOmit := omitOrMask(ctx, logger, &msg, additionalFields)
+	if shouldOmit {
+		return
+	}
+
+	logger.Error(msg, additionalArgs...)
+}
+
+func omitOrMask(ctx context.Context, logger hclog.Logger, msg *string, additionalFields []map[string]interface{}) ([]interface{}, bool) {
+	tfLoggerOpts := logging.GetProviderRootTFLoggerOpts(ctx)
+	additionalArgs := hclogutils.MapsToArgs(additionalFields...)
+	impliedArgs := logger.ImpliedArgs()
+
+	// Apply the provider root LoggerOpts to determine if this log should be omitted
+	if tfLoggerOpts.ShouldOmit(msg, impliedArgs, additionalArgs) {
+		return nil, true
+	}
+
+	// Apply the provider root LoggerOpts to apply masking to this log
+	tfLoggerOpts.ApplyMask(msg, impliedArgs, additionalArgs)
+	return additionalArgs, false
+}
+
+// WithOmitLogByKeys returns a new context.Context that has a modified logger
+// that will omit to write any log when any of the given keys is found
+// within the arguments.
+//
+// Each call to this function is additive:
+// the keys to omit by are added to the existing configuration.
+//
+// Example:
+//
+//   configuration = `['foo', 'baz']`
+//
+//   log1 = `{ msg = "...", args = { 'foo', '...', 'bar', '...' }`   -> omitted
+//   log2 = `{ msg = "...", args = { 'bar', '...' }`                 -> printed
+//   log3 = `{ msg = "...", args = { 'baz`', '...', 'boo', '...' }`  -> omitted
+//
+// Note that the key matching is a case-insensitive string comparison.
```

Comment:
Felt sufficient to catch most situations. Of course, someone might want to use the same key in different cases, and this would fall short.

I could refactor to be case-sensitive and call it a day.

Reply:
Refactored to omit this behaviour.